components/push/android/src/main/java/mozilla/appservices/push/PushManager.kt

```diff
@@ -228,24 +228,81 @@ class DispatchInfo constructor (
 
 /**
  * An API for interacting with Push.
+
+    Usage:
+
+    The push component is designed to be as light weight as possible. The "Push Manager"
+    handles subscription management and message decryption.
+
+    In general, usage would consist of calling:
+
+    ```kotlin
+    val manager = PushManager(
+        senderId = "SomeSenderIDValue",
+        bridgeType = BridgeType.FCM,
+        registrationId = systemProvidedRegistrationValue,
+        databasePath = "/path/to/database.sql"
+    )
+    val newEndpoints = manager.verifyConnection()
+    if newEndpoints.length() > 0 {
+        for (channelId in newEndpoints.keys()) {
+            // send the endpoint (newEndpoint[channelId]) to the process tied to channelId
+        }
+    }
+
+    // On new message:
+    // A new incoming message generally has the following format:
+    // {"chid": ChannelID, "body": Body, "con": Encoding, "enc": Salt, "crypto_key": DH}
+
+    val decryptedMessage = manager.decrypt(
+        channelID=message["chid"],
+        body=message["body"],
+        encoding=message["con"],
+        salt=message.getOrElse("enc", ""),
```

Comment:
encoding comes from "con", and salt comes from "enc". Is that correct?

Reply:
yes. This is a holdover from the previous version.

FWIW: 'enc' is short for the "Encryption" header, and 'con' for the "Content-Encoding" header.